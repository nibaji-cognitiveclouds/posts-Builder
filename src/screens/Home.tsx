/** @format */

import React from "react";
import { ActivityIndicator, FlatList, Text, View } from "react-native";
import { http } from "../const/https";

const Home = () => {
	const [data, setData] = React.useState([]);
	const [page, setPage] = React.useState(0);
	const [showHeaderLoader, setShowHeaderLoader] = React.useState(false);
	const [showFooterLoader, setShowFooterLoader] = React.useState(false);

	React.useEffect(() => {
		getData();
		theTimer(true);
	}, []);

	function getData(toTop: Boolean = false) {
		http
			.get(`v1/search_by_date?tags=story&page=${page}`)
			.then((res) => {
				const theArray = toTop
					? Array.from(new Set([...res.data.hits, ...data]))
					: Array.from(new Set([...data, ...res.data.hits]));
				setData(theArray);
			})
			.finally(() => {
				setShowHeaderLoader(false);
				setShowFooterLoader(false);
			});
	}

	function incrementPage() {
		setPage(page + 1);
	}

	function theTimer(toTop: Boolean = false) {
		setInterval(() => {
			setShowHeaderLoader(true);
			incrementPage();
			getData(toTop);
		}, 10000);
	}
	return (
		<View>
			<FlatList
				data={data}
				ListHeaderComponent={() => {
					if (showHeaderLoader) {
						return <ActivityIndicator />;
					} else {
						return null;
					}
				}}
				renderItem={({ item }) => (
					<Text
						style={{ margin: 5, padding: 5, backgroundColor: "rgba(0,0,0,.2)" }}
					>
						{item?.title}
					</Text>
				)}
				onEndReached={() => {
					incrementPage();
					setShowFooterLoader(true);
					getData();
				}}
				ListFooterComponent={() => {
					if (showFooterLoader) {
						return <ActivityIndicator />;
					} else {
						return null;
					}
				}}
			/>
		</View>
	);
};

export default Home;
