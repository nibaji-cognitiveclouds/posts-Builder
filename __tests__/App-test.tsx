/**
 * @format
 */

import "react-native";
import React from "react";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import Home from "../src/screens/Home";

describe("Home", () => {
	it("renders <Home /> correctly", () => {
		jest.useFakeTimers();
		renderer.create(<Home />);
	});

	it("rendered <Home /> matches snapshot", () => {
		jest.useFakeTimers();
		const render = renderer.create(<Home />);
		expect(render).toMatchSnapshot();
	});
});
