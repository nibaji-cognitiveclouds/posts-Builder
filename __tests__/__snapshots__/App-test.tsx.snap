// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home rendered <Home /> matches snapshot 1`] = `
<View>
  <RCTScrollView
    ListFooterComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      />
      <View
        onLayout={[Function]}
      />
    </View>
  </RCTScrollView>
</View>
`;
